napari/plugins/_npe2.py

```diff
@@ -364,20 +423,34 @@ def _npe2_manifest_to_actions(
                     subitem = _npe2_submenu_to_app_model(item)
                     submenus.append((menu_id, subitem))
 
-    actions: List[Action] = [
-        Action(
-            id=cmd.id,
-            title=cmd.title,
-            category=cmd.category,
-            tooltip=cmd.short_title or cmd.title,
-            icon=cmd.icon,
-            enablement=cmd.enablement,
-            callback=cmd.python_name or '',
-            menus=cmds.get(cmd.id),
-            keybindings=[],
-        )
-        for cmd in mf.contributions.commands or ()
-    ]
+    if mf.contributions.commands:
+        data_commands = [
+            c.command
+            for _, contribs in pm.iter_sample_data()
+            for c in contribs
+        ]
```

Comment:
AFAICT `pm.iter_sample_data()` will give all sample data commands. Thus it may be minimally faster/better (?) to get just the sample data contributions of the current manifest (iterating through `mf.contributions.sample_data` and getting the `command` property)

Happy to change.